### PR TITLE
Fix static check for vendor/k8s.io/apiserver/pkg/storage/value/encrypt/envelope

### PR DESCRIPTION
**Pull request by CaoDonghui123**
**What type of PR is this?**

/kind cleanup
**What this PR does / why we need it**:
static check on vendor/k8s.io/apiserver/pkg/storage/value/encrypt/envelope 
**Which issue(s) this PR fixes**:

Part of #92402

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/grpc_service_unix_test.go [hunk 1]
```diff
@@ -138,7 +138,7 @@ func TestTimeouts(t *testing.T) {
 
 				service, err = NewGRPCService(socketName.endpoint, tt.callTimeout)
 				if err != nil {
-					t.Fatalf("failed to create envelope service, error: %v", err)
+					t.Errorf("failed to create envelope service, error: %v", err)
```
Comment (deads2k) [inline staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/grpc_service_unix_test.go:141]: this test seems like an ideal place to nest, defer the `wg.Done`, pipe errors to a channel, check the channel after the wait.

File: staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/grpc_service_unix_test.go [hunk 2]
```diff
@@ -228,7 +228,7 @@ func TestIntermittentConnectionLoss(t *testing.T) {
 		wg1.Done()
 		_, err := service.Encrypt(data)
 		if err != nil {
-			t.Fatalf("failed when executing encrypt, error: %v", err)
+			t.Errorf("failed when executing encrypt, error: %v", err)
```
Comment (deads2k) [inline staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/grpc_service_unix_test.go:231]: same here

File: staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/grpc_service_unix_test.go (hunk 1 above)
Comment (CaoDonghui123) [reply staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/grpc_service_unix_test.go:141]: > this test seems like an ideal place to nest, defer the `wg.Done`, pipe errors to a channel, check the channel after the wait.

I edit it just fix "SA2002 | Called testing.T.FailNow or SkipNow in a goroutine, which isn't allowed"

Comment (CaoDonghui123) [reply staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/grpc_service_unix_test.go:141]: > this test seems like an ideal place to nest, defer the `wg.Done`, pipe errors to a channel, check the channel after the wait.

 I get it

File: staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/grpc_service_unix_test.go [hunk 3]
```diff
@@ -138,7 +138,8 @@ func TestTimeouts(t *testing.T) {
 
 				service, err = NewGRPCService(socketName.endpoint, tt.callTimeout)
 				if err != nil {
-					t.Fatalf("failed to create envelope service, error: %v", err)
+					t.Errorf("failed to create envelope service, error: %v", err)
+					return
```
Comment (deads2k) [inline staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/grpc_service_unix_test.go:142]: by returning before the .Done is called, you can permawedge this test case.

**Pull request closed without merging**